### PR TITLE
Fix reward dtype in FourRooms

### PR DESCRIPTION
**Pull request by jiriyu98**
Based on type annotation and according to tradition, I think this behavior is unexpected.

jnp.logical_and won't convert reward dtype from bool to float. Simply add `*1.0` to convert it.

### CODE REVIEW

File: gymnax/environments/misc/rooms.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def step_env(
         new_pos = jax.lax.select(in_map, p, state.pos)
         reward = jnp.logical_and(
             new_pos[0] == state.goal[0], new_pos[1] == state.goal[1]
-        )
+        ) * 1.0
```
Comment (dantp-ai) [inline gymnax/environments/misc/rooms.py:108]: nit: looks good. I was wondering why you didn't decide to use `jax.numpy.astype()` for this?

**Review by dantp-ai — APPROVED** [review]

**Comment by jiriyu98** [comment]
sorry I just came back and found that there was already a fix - so let me close this pr. ;-)
